tests/Operation/RenameCollectionFunctionalTest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\Driver\Exception\CommandException;
+use MongoDB\Operation\DropCollection;
+use MongoDB\Operation\FindOne;
+use MongoDB\Operation\InsertOne;
+use MongoDB\Operation\RenameCollection;
+use MongoDB\Tests\CommandObserver;
+
+use function version_compare;
+
+class RenameCollectionFunctionalTest extends FunctionalTestCase
+{
+    /** @var integer */
+    private static $errorCodeNamespaceNotFound = 26;
+
+    /** @var integer */
+    private static $errorCodeNamespaceExists = 48;
+
+    /** @var string */
+    private $toCollectionName;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->toCollectionName = $this->getCollectionName() . '.renamed';
+        $operation = new DropCollection($this->getDatabaseName(), $this->toCollectionName);
+        $operation->execute($this->getPrimaryServer());
+    }
+
+    public function tearDown(): void
+    {
+        if ($this->hasFailed()) {
+            return;
+        }
+
+        $operation = new DropCollection($this->getDatabaseName(), $this->toCollectionName);
+        $operation->execute($this->getPrimaryServer());
+
+        parent::tearDown();
+    }
+
+    public function testDefaultWriteConcernIsOmitted(): void
+    {
+        (new CommandObserver())->observe(
+            function (): void {
+                $server = $this->getPrimaryServer();
+
+                $insertOne = new InsertOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1]);
+                $writeResult = $insertOne->execute($server);
+                $this->assertEquals(1, $writeResult->getInsertedCount());
+
+                $operation = new RenameCollection(
+                    $this->getDatabaseName(),
+                    $this->getCollectionName(),
+                    $this->getDatabaseName(),
+                    $this->toCollectionName,
+                    ['writeConcern' => $this->createDefaultWriteConcern()]
+                );
+
+                $operation->execute($server);
+            },
+            function (array $event): void {
+                $this->assertObjectNotHasAttribute('writeConcern', $event['started']->getCommand());
+            }
+        );
+    }
+
+    public function testRenameExistingCollection(): void
```

Comment:
```suggestion
    public function testRenameCollectionToNonexistentTarget(): void
```

A nonexistent source collection is an exception, and covered in a later test that expects an error. The successful path involves a collection that _does_ exist, so I don't think we need to clarify that. The distinguishing characteristic of this test is that the target does not exist.